planning/behavior_velocity_planner/src/scene_module/intersection/util.cpp

```diff
@@ -351,72 +360,49 @@ bool getObjectiveLanelets(
   const auto & conflicting_lanelets =
     lanelet::utils::getConflictingLanelets(routing_graph_ptr, assigned_lanelet);
 
-  std::vector<lanelet::ConstLanelets>                      // conflicting lanes with "lane_id"
-    conflicting_lanelets_ex_yield_ego;                     // excluding ego lanes and yield lanes
-  std::vector<lanelet::ConstLanelets> objective_lanelets;  // final objective lanelets
+  lanelet::ConstLanelets detection_lanelets;  // final objective lanelets
 
-  // exclude yield lanelets and ego lanelets from objective_lanelets
-  for (const auto & conflicting_lanelet : conflicting_lanelets) {
-    if (lanelet::utils::contains(yield_lanelets, conflicting_lanelet)) {
-      continue;
-    }
-    if (lanelet::utils::contains(ego_lanelets, conflicting_lanelet)) {
-      continue;
+  // exclude yield lanelets and ego lanelets from detection_lanelets
+  // if assigned lanelet is "straight" with traffic light, detection area is not necessary
+  if (turn_direction == std::string("straight") && has_tl) {
```

Comment:
Is this condition necessary?
Explicitly indicate detection are is not necessary under this consition?

Reply:
@soblin 
How about this comment?